[16.0][ADD] helpdesk_mgmt_purchase: links purchase orders and helpdesk tickets.

nicolas-delbovier-acsone opened the pull request:
<img width="1370" height="643" alt="image" src="https://github.com/user-attachments/assets/24a3e9c8-a905-4715-a57d-d9dcae9d5dc1" />

<img width="1523" height="654" alt="image" src="https://github.com/user-attachments/assets/863e932e-7bdb-4ef6-8d67-5ef21c5bb2f7" />

nicolas-delbovier-acsone committed:
[ADD] helpdesk_mgmt_purchase: links purchase orders and helpdesk tickets.

nicolas-delbovier-acsone committed:
[ADD] helpdesk_mgmt_purchase: links purchase orders and helpdesk tickets.

lmignon commented on helpdesk_mgmt_purchase/models/purchase_order.py:
```diff
@@ -0,0 +1,19 @@
+    @api.depends("ticket_ids")
+    def _compute_ticket_count(self):
+        for order in self:
+            order.ticket_count = len(order.ticket_ids)
```
@nicolas-delbovier-acsone you should use a `read_group`to avoid perf issue if the field is accessed on a large recordsed....

nihelgabsi-acsone committed:
[16.0][IMP] helpdesk_mgmt_purchase: use read_group method for count fields to improve perf

nicolas-delbovier-acsone committed:
[FIX] helpdesk_mgmt_purchase: remove incorrect Sale Order context

The button to view Helpdesk tickets from a Purchase Order was incorrectly
passing a 'default_sale_order_id' in the action context due to a
copy-paste error from the sales module.

This caused new tickets created from a PO view to be linked to an
unrelated Sale Order.

nicolas-delbovier-acsone committed:
[REF] helpdesk_mgmt_purchase: move open tickets action to Python

Shifting the action logic to Python facilitates dynamic customization. This approach allows inheriting modules to override or extend the action via super(), avoiding the limitations and complexity of XML xpath inheritance for conditional logic

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_purchase: enhance action context validation in tests

Following the removal of the erroneous 'default_sale_order_id' from
the purchase action, this commit strengthens the test suite to
prevent regressions.

The test logic now explicitly validates the content of the action
context. This ensures that:
- The expected Purchase Order ID is present.
- No unrelated context keys (like 'default_sale_order_id') are
  leaking into the view from inherited methods or copy-paste errors.

lmignon reviewed (changes requested):
@nicolas-delbovier-acsone please make sure to fill in the README.rst files with the module info and remove any empty ones.

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_purchase: clean readme files

nicolas-delbovier-acsone committed:
[IMP] helpdesk_mgmt_purchase: improve unit tests

rousseldenis commented on helpdesk_mgmt_purchase/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+    "version": "16.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "ACSONE SA/NV,Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/helpdesk",
```
@nicolas-delbovier-acsone Don't hesitate to adopt this module as maintainer 😃
